admin/src/js/controllers/forms-xml.js

```diff
@@ -92,21 +100,24 @@ angular.module('controllers').controller('FormsXmlCtrl',
             );
           }
 
-          const couchId = 'form:' + formId;
-          return DB().get(couchId, { include_attachments:true })
-            .catch(function(err) {
-              if (err.status === 404) {
-                return { _id: couchId };
-              }
-              throw err;
-            })
-            .then(function(doc) {
-              doc.title = title;
-              Object.assign(doc, meta);
-              doc.type = 'form';
-              doc.internalId = formId;
-              AddAttachment(doc, 'xml', xml, 'application/xml');
-              return doc;
+          return ValidateForm(xml)
+            .then(() => {
+              const couchId = 'form:' + formId;
```

Comment:
I know there are no unit tests for this file, but I think this new interaction deserves some unit testing.